### PR TITLE
KAFKA-4013: Included exception cause in SaslServerCallbackHandler

### PR DESCRIPTION
**Pull request by bbaugher**

### CODE REVIEW

File: clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerCallbackHandler.java [hunk 1]
```diff
@@ -89,7 +89,7 @@ private void handleAuthorizeCallback(AuthorizeCallback ac) {
             LOG.info("Setting authorizedID: {}", userName);
             ac.setAuthorizedID(userName);
         } catch (IOException e) {
-            LOG.error("Failed to set name based on Kerberos authentication rules.");
+            LOG.error("Failed to set name for [{}] based on Kerberos authentication rules.", kerberosName, e);
```
Comment (ijuma) [inline clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerCallbackHandler.java]: We don't typically use `[]` in this way, maybe single quotes instead?

Comment (bbaugher) [inline clients/src/main/java/org/apache/kafka/common/security/authenticator/SaslServerCallbackHandler.java]: Sorry habit of how I've done things. Done

**Comment by harshach** [comment]
+1. Build failure not related to the patch.

**Comment by ijuma** [comment]
Thanks for the PR, LGTM.

**Pull request closed without merging**